beets/util/__init__.py

```diff
@@ -442,17 +450,29 @@ def samefile(p1, p2):
     return shutil._samefile(syspath(p1), syspath(p2))
 
 
-def remove(path, soft=True):
-    """Remove the file. If `soft`, then no error will be raised if the
-    file does not exist.
+def remove(path, soft=True, remove_command=config["remove_command"].as_str()):
+    """Remove file via command specified in the `remove_command` setting if
+    specified executable is found on $PATH. Otherwise, delete the file.
```

Comment:
Maybe it would be safer to raise an Exception when the executable cannot be found? It appears to me that currently, a typo in the `remove_command` option would be completely silent. The same would happen by accidentally uninstalling the trash program.

Reply:
The idea is to provide — without any action on the part of end users — a sane, safe default, which is to use the `trash` executable if present. If not present, the current behavior gracefully falls back to the pre-existing `os.remove()` behavior. If an exception were raised when the executable isn't found, then the first-run experience for anyone without a `trash` executable on `$PATH` is going to see a surprising exception that they may not understand. I don't think that's a great user experience.

It's very easy to tell whether files are being moved to trash or not, so I don't think silently falling back to the pre-existing `os.remove()` behavior is problematic.

Reply:
Right - I hadn't realised that the default would become `remove_command=trash` rather than `remove_command=""`. If using `trash` is going to be the default, you're right of course.